server/src/test/java/org/opensearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1780,6 +1781,48 @@ public void testImproveConfigurationPerformsVotingConfigExclusionStateCheck() {
         }
     }
 
+    public void testLocalNodeAlwaysCommissionedWithoutDecommissionedException() {
+        try (Cluster cluster = new Cluster(randomIntBetween(1, 5))) {
+            cluster.runRandomly();
+            cluster.stabilise();
+            for (ClusterNode node : cluster.clusterNodes) {
+                assertTrue(node.coordinator.localNodeCommissioned());
+            }
+        }
+    }
+
+    public void testClusterStabilisesAfterDecommissionedIntervalForPreviouslyDecommissionedNode() {
```

Comment:
More tests around:
1. Decommissioned node is not able to join - both cluster manager eligible or data node
2. Decommissioned node doesn't become leader (by triggering election)

Reply:
test rename to ```testClusterStabilisesForPreviouslyDecommissionedNode```